src/core/namespace.c

```diff
@@ -88,6 +88,7 @@ typedef struct MountEntry {
         bool noexec:1;            /* Shall set MS_NOEXEC on the mount itself */
         bool exec:1;              /* Shall clear MS_NOEXEC on the mount itself */
         bool applied:1;           /* Already applied */
+        bool skip_postprocessing:1; /* Avoid remounting (r/o, nodev, nosuid) */
```

Comment:
where's the difference to `applied`?

Reply:
stuff like /, /proc, /dev and so on still needs to be processed even if it is skipped

Reply:
i don't understand?

Reply:
Check the reproducer in the test case - when a BindReadOnlyPath=-/foo:/usr is given and /foo doesn't exist, it is silently skipped as expected, but then /usr is remounted read-only anyway, because this doesn't take into account that skipped mounts. This fixes that, so that we know what we skip and thus shouldn't have further processing, and what should always have further processing - like ProtectSystem= for example, which uses mounts

Reply:
hmm, so i read the code now. "applied" means we looked at an entry already, it doesn't mean we actually applied it, we might have skipped over it, because it didn't work. I think this field should become an enum:

```c
typedef enum MountEntryState {
      MOUNT_UNAPPLIED,
      MOUNT_APPLIED,
      MOUNT_SKIPPED,
      _MOUNT_ENTRY_STATE_MAX,
      _MOUNT_ENTRY_STATE_INVALID = -EINVAL,
} MountEntryState; 
```

And then we should skip the "post-processing" if the state is not MOUNT_APPLIED.